deeppavlov/dataset_readers/imdb_reader.py

```diff
@@ -0,0 +1,76 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from logging import getLogger
+from typing import List, Dict, Any, Optional, Tuple
+from pathlib import Path
+
+from overrides import overrides
+
+from deeppavlov.core.common.registry import register
+from deeppavlov.core.data.dataset_reader import DatasetReader
+from deeppavlov.core.data.utils import download_decompress, mark_done, is_done
+
+log = getLogger(__name__)
+
+
+@register('imdb_reader')
+class ImdbReader(DatasetReader):
+    """This class downloads and reads the IMDb sentiment classification dataset.
+
+    https://ai.stanford.edu/~amaas/data/sentiment/
+
+    Andrew L. Maas, Raymond E. Daly, Peter T. Pham, Dan Huang, Andrew Y. Ng, and Christopher Potts.
+    (2011). Learning Word Vectors for Sentiment Analysis. The 49th Annual Meeting of the Association
+    for Computational Linguistics (ACL 2011).
+    """
+
+    @overrides
+    def read(self, data_path: str, url: Optional[str] = None,
+             *args, **kwargs) -> Dict[str, List[Tuple[Any, Any]]]:
+        """
+        Args:
+            data_path: A path to a folder with dataset files.
+            url: A url to the archive with the dataset to download if the data folder is empty.
+        """
+        data_path = Path(data_path)
+
+        if url is None:
+            url = "http://ai.stanford.edu/~amaas/data/sentiment/aclImdb_v1.tar.gz"
```

Comment:
hidden logic, why should the file be available at this link?

Reply:
Please suggest how to fix.

Reply:
Now the path must be specified directly in the config. Add "url" to  "dataset_reader": {..}

Reply:
I'm not convinced, many dataset readers of DeepPavlov have their URLs hardcoded, and forcing the user to provide the URL in the config file will harm usability. (However almost all of these harcoded URLs lead to `files.deeppavlov.ai/*`, so the imdb dataset should probably be uploaded to `files.deeppavlov.ai` as well).

Reply:
@yoptar what do you think about this? Maybe, i was wrong